lib/events.js

```diff
@@ -101,18 +103,36 @@ function emitThree(handler, isFn, self, arg1, arg2, arg3) {
   }
 }
 
+function emitMany(handler, isFn, self, args) {
+  if (isFn)
+    handler.apply(self, args);
+  else {
+    var len = handler.length;
+    var listeners = arrayClone(handler, len);
+    for (var i = 0; i < len; ++i)
+      listeners[i].apply(self, args);
+  }
+}
+
 EventEmitter.prototype.emit = function emit(type) {
-  var er, handler, len, args, i, listeners, events, domain;
+  var er, handler, len, args, i, events, domain;
   var needDomainExit = false;
+  var doError = false;
 
   events = this._events;
-  if (!events)
+  if (!events) {
     events = this._events = {};
+    this._eventsCount = 0;
+    if (type === 'error')
+      doError = true;
+    else
+      return false;
+  }
```

Comment:
Suggestion: perhaps you can rewrite this so that it doesn't mutate `this` when `type === 'error' && events == null`?  You could drop the `doError` local if you change the check below to:

```javascript
if (type === 'error' && (events == null || events.error == null)) {
```